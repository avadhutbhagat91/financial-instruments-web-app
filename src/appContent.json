{
    "financialInstrumentsContent":{
      "header": "Financial Instruments Assignment",
      "ticker": "Ticker",
      "price": "Price",
      "assetClass": "Asset Class",
      "tickerToolTip": "Click here to sort table by ticker data alphabetically.",
      "priceToolTip": "Click here to sort table by price in descending order.",
      "assetClassToolTip": "Click here to sort table by asset class with Equities first then Macro then Credit."
    }
  }