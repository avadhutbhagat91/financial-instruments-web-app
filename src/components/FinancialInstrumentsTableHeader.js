import { PRICE, ASSET_CLASS, TICKER } from '../appConstants.js';

const FinancialInstrumentsTableHeader = ({
  appContent,
  sortFinancialInstruments
}) => {
  const {
    financialInstrumentsContent: {
      ticker,
      price,
      assetClass,
      tickerToolTip,
      priceToolTip,
      assetClassToolTip
    }
  } = appContent;
  return (
    <thead>
      <tr>
        <th
          onClick={() => sortFinancialInstruments(TICKER)}
          title={tickerToolTip}
          className="header"
        >
          {ticker}
        </th>
        <th
          onClick={() => sortFinancialInstruments(PRICE)}
          title={priceToolTip}
          className="header"
        >
          {price}
        </th>
        <th
          onClick={() => sortFinancialInstruments(ASSET_CLASS)}
          title={assetClassToolTip}
          className="header"
        >
          {assetClass}
        </th>
      </tr>
    </thead>
  );
};
export default FinancialInstrumentsTableHeader;