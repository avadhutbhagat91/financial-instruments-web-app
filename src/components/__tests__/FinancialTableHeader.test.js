import React from 'react';
import { shallow } from 'enzyme';
import FinancialInstrumentsTableHeader from '../FinancialInstrumentsTableHeader';
import financialInstruments from '../../financialInstruments.json';
import appContent from '../../appContent.json'

const setup = (setupProps = {}) => {
    const defaultProps = {
        financialInstruments: financialInstruments
    }
    const props = { ...defaultProps, ...setupProps};
    const wrapper = shallow(<FinancialInstrumentsTableHeader {...props}/>);
    return {
        props,
        wrapper
    }
}

describe('FinancialInstrumentsTableHeader', () => {
    test('render', () => {
        const { wrapper } = setup({ appContent: appContent });
        expect(wrapper).toMatchSnapshot();
    });
});