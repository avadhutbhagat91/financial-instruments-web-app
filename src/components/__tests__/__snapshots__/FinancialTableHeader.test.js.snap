// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancialInstrumentsTableHeader render 1`] = `
<thead>
  <tr>
    <th
      className="header"
      onClick={[Function]}
      title="Click here to sort table by ticker data alphabetically."
    >
      Ticker
    </th>
    <th
      className="header"
      onClick={[Function]}
      title="Click here to sort table by price in descending order."
    >
      Price
    </th>
    <th
      className="header"
      onClick={[Function]}
      title="Click here to sort table by asset class with Equities first then Macro then Credit."
    >
      Asset Class
    </th>
  </tr>
</thead>
`;
